pkg/persistence/redis/redisclient.go

```diff
@@ -0,0 +1,160 @@
+package redis
+
+import (
+"errors"
+"fmt"
+"net/url"
+"strconv"
+"strings"
+"sync"
+"time"
+
+"github.com/FZambia/sentinel"
+"github.com/gomodule/redigo/redis"
+)
+
+var knownPool sync.Map
```

Comment:
What is the use case for having more than one pool? Is see that we only construct and use the pool named `Default`

Reply:
The code is I copied from harbor just want to keep them same.
However we can just make it as a singleton.

Reply:
Refactored to keep only one instance of redis.Pool, so the code is also simpler